api/metrics/legacy/http.go

```diff
@@ -141,23 +141,32 @@ func NewHandler(url *url.URL, upstreamCA []byte, upstreamCert *stdtls.Certificat
 	}
 
 	r.Group(func(r chi.Router) {
+		r.Use(func(handler http.Handler) http.Handler {
```

Comment:
The instrumentation middleware is run first (line 189), then the next middleware injects the extra labels `group` and `handler` into context so that when the middleware for ratelimiter runs and when a 429 is returned back to the instrumentation handler(here -> [line101](https://github.com/observatorium/api/blob/56abe030cbbe3e50a614dacf793fa673317b937b/server/http_instrumentation.go#L101)), the ctx already has the labels for `group` and `handler` set

Reply:
There is a design problem in injecting these labels here: the handler is not decided yet. 

For example, we could have the following: 

```go
		r.Group(func(r chi.Router) {
			r.Use(func(handler http.Handler) http.Handler {
				return server.InjectLabelsCtx(
					prometheus.Labels{"group": "metricsv1", "handler": "query"},
					handler,
				)
			})
			r.Use(c.queryMiddlewares...)
			r.Use(server.StripTenantPrefix("/api/metrics/v1"))
			r.Handle(QueryRoute,
				otelhttp.WithRouteTag(
					c.spanRoutePrefix+QueryRoute,
					proxyQuery,
				),
			)
            // A different query route -- maybe a different type of query besides range and instant. Or simply a new range query endpoint with a slightly different middleware stack.
			r.Handle(DifferentQueryRoute, 
				otelhttp.WithRouteTag(
					c.spanRoutePrefix+QueryRoute,
					differentQueryProxy,
				),
			)
		})
```

With the changes in this PR both routes would report the same group and handler, but in fact they are different handlers.

Reply:
The `handler` label can only be injected when the request arrives at the "final" `http.Handler`, often inside a `router.Handle` call. Otherwise we will need a whole new `router.Group` if we add a new route here.

Reply:
Ah, we already have them separate anyway, so no problem. 😅